web/src/lib/constants.ts

```diff
@@ -59,6 +60,32 @@ export const dateFormats = {
   },
 };
 
+export enum QueryParameter {
```

Comment:
In the future, this would be a good change to do in a separate pull request. If you find yourself refactoring stuff in order to make the current task easier, considering doing it in a separate pull request, as it is easier to review, test, etc. (easier to review means it gets merged faster :smile:)

Reply:
Refactoring is too good :stuck_out_tongue_winking_eye: 

All joking aside, yes I should keep separate changes in separate PRs, but every time I start refactoring something I'm like "it's a small change, it's not worth a whole PR for so little change"

For this case, I limited myself to "only" change the way query parameters are handled. I plan to refactor the routing through the webapp.